### PR TITLE
LPS-130855 Validate input mask when saving the form

### PR DESCRIPTION
**Pull request by carolmariaabb**
Related Issue: https://issues.liferay.com/browse/LPS-130855

### CODE REVIEW

**Comment by liferay-forms** [comment]
Hi, you don’t need to run the CI commands anymore, I’ll do it for you and let you know if any issue happens.

**Comment by liferay-forms** [comment]
ci:test:sf

**Comment by liferay-forms** [comment]
ci:test:relevant

**Review by natocesarrego — APPROVED** [review]

**Comment by natocesarrego** [comment]
@carolmariaabb, this pull is ready to be forwarded, but, from what I could understand, it depends of https://github.com/natocesarrego/liferay-portal/pull/995, therefore I'll wait the other pull be forwarded before move on with this one. If my understanding is incorrect, please feel free to forward this pull.

Thanks.

File: modules/apps/dynamic-data-mapping/dynamic-data-mapping-form-evaluator-impl/src/main/java/com/liferay/dynamic/data/mapping/form/evaluator/internal/helper/DDMFormEvaluatorHelper.java [hunk 1]
```diff
@@ -560,7 +558,73 @@ protected void updateDDMFormFieldValue(
 		value.addString(locale, String.valueOf(newValue));
 	}
 
-	protected void validateField(
+	protected void validateFields() {
+		validateFieldsMarkedAsRequired();
+
+		validateFieldsWithConfirmationField();
+
+		validateFieldsWithDDMFormFieldValidation();
```
[the remark this answers is not in the document]
Comment (carolmariaabb) [reply modules/apps/dynamic-data-mapping/dynamic-data-mapping-form-evaluator-impl/src/main/java/com/liferay/dynamic/data/mapping/form/evaluator/internal/helper/DDMFormEvaluatorHelper.java:602]: Thanks 🎉

File: modules/apps/dynamic-data-mapping/dynamic-data-mapping-form-evaluator-impl/src/main/java/com/liferay/dynamic/data/mapping/form/evaluator/internal/helper/DDMFormEvaluatorHelper.java [hunk 2]
```diff
@@ -754,17 +754,10 @@ protected void validateFieldWithDDMFormFieldValidation(
...
-
-			GetFieldPropertyResponse getFieldPropertyResponse =
-				ddmFormEvaluatorDDMExpressionFieldAccessor.getFieldProperty(
-					builder.build());
-
 			ddmExpression.setVariable(
-				fieldName, getFieldPropertyResponse.getValue());
+				fieldName,
+				_getFieldPropertyResponseValue(
+					ddmFormEvaluatorFieldContextKey, "value"));
```
[the remark this answers is not in the document]
Comment (carolmariaabb) [reply modules/apps/dynamic-data-mapping/dynamic-data-mapping-form-evaluator-impl/src/main/java/com/liferay/dynamic/data/mapping/form/evaluator/internal/helper/DDMFormEvaluatorHelper.java:760]: Thanks 😄

**Comment by carolmariaabb** [comment]
[#995](https://github.com/natocesarrego/liferay-portal/pull/995) was merged by Brian. Therefore, I will forward this pull.

**Comment by carolmariaabb** [comment]
ci:forward

**Pull request closed without merging**